tests/integration/test_dynamodb.py

```diff
@@ -380,6 +380,35 @@ def test_create_duplicate_table(self):
             )
         self.assertIn('ResourceInUseException', str(ctx.exception))
 
+        # clean up
+        delete_table(table_name)
+
+    def test_delete_table(self):
+        table_name = 'test-ddb-table-%s' % short_uid()
+        dynamodb = aws_stack.connect_to_service('dynamodb')
+
+        dynamodb.create_table(
+            TableName=table_name,
+            KeySchema=[{
+                'AttributeName': 'id', 'KeyType': 'HASH'
+            }],
+            AttributeDefinitions=[{
+                'AttributeName': 'id', 'AttributeType': 'S'
+            }],
+            ProvisionedThroughput={
+                'ReadCapacityUnits': 1, 'WriteCapacityUnits': 1
+            },
+        )
+
+        table_list = dynamodb.list_tables()
+        self.assertEqual(1, len(table_list['TableNames']))
```

Comment:
Looks like we're currently seeing a test error here:

```
======================================================================

FAIL: test_delete_table (tests.integration.test_dynamodb.DynamoDBIntegrationTest)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/opt/code/localstack/tests/integration/test_dynamodb.py", line 404, in test_delete_table

    self.assertEqual(1, len(table_list['TableNames']))

AssertionError: 1 != 2

-------------------- >> begin captured logging << --------------------
```

I think we should first get the number of tables at the beginning of the test
```
tables_before = len(dynamodb.list_tables()['TableNames'])
...
```
... and then assert that the number increased by one:
```
...
self.assertEqual(tables_before + 1, len(table_list['TableNames']))
```